### PR TITLE
OSDOCS-11417#Reworking ASH topic map/TOC, creating IPI and UPI directories, updating 'Preparing to install a cluster' assembly

### PR DESCRIPTION
**Pull request by brendan-daly-red-hat**
Versions:
4.17+

Issue:
https://issues.redhat.com/browse/OSDOCS-11417

Link to docs preview:

- [Installation methods](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/preparing-to-install-on-azure-stack-hub)
- [Preparing to install a cluster on Azure Stack Hub](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/ipi/ipi-ash-preparing-to-install) (IPI)
- [Installing a cluster on Azure Stack Hub with customizations](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-default)
- [Installing a cluster on Azure Stack Hub with network customizations](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-network-customizations)
- [Preparing to install a cluster on Azure Stack Hub](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/upi/upi-ash-preparing-to-install) (UPI)
- [Installing a cluster on Azure Stack Hub using ARM templates](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/upi/installing-azure-stack-hub-user-infra)

Summary of changes:

- Create new ‘Installer-provisioned infrastructure’ and ‘User-provisioned infrastructure’ directories and fix relative paths for xrefs
- Remove ‘Azure Stack Hub’ text from most of the topic_maps file (i.e., this text is removed from the TOC, but retained in the page titles)
- Remove five modules (cluster-entitlements.adoc, ssh-agent-using.adoc, installation-obtaining-installer.adoc, cli-installing-cli.adoc, and cluster-telemetry.adoc) from installing-azure-stack-hub-default.adoc and from installing-azure-stack-hub-network-customizations.adoc
- Add the five modules to ipi-ash-preparing-to-install.adoc 
- Remove four modules (cluster-entitlements.adoc, ssh-agent-using.adoc, installation-obtaining-installer.adoc, and cli-installing-cli.adoc) from installing-azure-stack-hub-user-infra.doc 
- Add the four modules to upi-ash-preparing-to-install.adoc 

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: installing/installing_azure_stack_hub/ipi/ipi-ash-preparing-to-install.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* Install Azure Stack Hub version 2008 or later.
+
+* Verify that you have approximately 16 GB of local disk space. Installing the cluster requires that you download the {op-system} virtual hard drive (VHD) cluster image and upload it to your Azure Stack Hub environment so that it is accessible during deployment. Decompressing the VHD files requires this amount of local disk space.
+
+In {product-title} version {product-version}, you can install a cluster on Microsoft Azure Stack Hub with an installer-provisioned infrastructure. However, you must manually configure the `install-config.yaml` file to specify values that are specific to Azure Stack Hub.
+
+[NOTE]
+====
+While you can select `azure` when using the installation program to deploy a cluster using installer-provisioned infrastructure, this option is only supported for the Azure Public Cloud.
+====
```
Comment (bscott-rh) [inline installing/installing_azure_stack_hub/ipi/ipi-ash-preparing-to-install.adoc:26]: I think these prerequisites are basically covered by the prerequisites in the [install assembly](https://80041--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-default#prerequisites_installing-azure-stack-hub-default-ipi) so they shouldn't be repeated here. Instead, this prerequisites section should list the steps that take place in the modules below, like we do [for AWS](https://docs.openshift.com/container-platform/4.16/installing/installing_aws/ipi/ipi-aws-preparing-to-install.html)

File: installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-default.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+.Additional resources
+* xref:../../../web_console/web-console.adoc#web-console[Accessing the web console]
+
+[id="next-steps_installing-azure-stack-hub-default-ipi"]
+== Next steps
+
+* xref:../../../installing/validation_and_troubleshooting/validating-an-installation.adoc#validating-an-installation[Validating an installation].
+* xref:../../../post_installation_configuration/cluster-tasks.adoc#available_cluster_customizations[Customize your cluster].
+* If necessary, you can xref:../../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.adoc#opting-out-remote-health-reporting_opting-out-remote-health-reporting[opt out of remote health reporting].
+* If necessary, you can xref:../../../post_installation_configuration/changing-cloud-credentials-configuration.adoc#manually-removing-cloud-creds_changing-cloud-credentials-configuration[remove cloud provider credentials].
```
Comment (adellape) [inline installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-default.adoc:61]: I don't think we have good guidance on exactly how ["Next steps"](https://redhat-documentation.github.io/modular-docs/#next-steps) sections should look / be punctuated, but if you wanted to make it adhere more to the "Additional resources" style (with no punctuation and minimal extra words), maybe something like:
```suggestion
* xref:../../../installing/validation_and_troubleshooting/validating-an-installation.adoc#validating-an-installation[Validating an installation]
* xref:../../../post_installation_configuration/cluster-tasks.adoc#available_cluster_customizations[Customize your cluster]
* Optional: xref:../../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.adoc#opting-out-remote-health-reporting_opting-out-remote-health-reporting[Opt out of remote health reporting]
* Optional: xref:../../../post_installation_configuration/changing-cloud-credentials-configuration.adoc#manually-removing-cloud-creds_changing-cloud-credentials-configuration[remove cloud provider credentials]
```

File: installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-network-customizations.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+[role="_additional-resources"]
+[id="additional-resources_installing-azure-stack-hub-network-customizations-console"]
+.Additional resources
+* xref:../../../web_console/web-console.adoc#web-console[Accessing the web console].
+
+[role="_additional-resources"]
+[id="additional-resources_installing-azure-stack-hub-network-customizations-telemetry"]
+
+[id="next-steps_installing-azure-stack-hub-network-customizations"]
+== Next steps
```
Comment (adellape) [inline installing/installing_azure_stack_hub/ipi/installing-azure-stack-hub-network-customizations.adoc:75]: Same as previous "Next steps" comment, if you so choose.

File: installing/installing_azure_stack_hub/ipi/ipi-ash-preparing-to-install.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+include::modules/installation-obtaining-installer.adoc[leveloffset=+1]
+
+include::modules/cli-installing-cli.adoc[leveloffset=+1]
+
+include::modules/cluster-telemetry.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* See xref:../../../support/remote_health_monitoring/about-remote-health-monitoring.adoc#about-remote-health-monitoring[About remote health monitoring] for more information about the Telemetry service.
```
Comment (adellape) [inline installing/installing_azure_stack_hub/ipi/ipi-ash-preparing-to-install.adoc:36]:
```suggestion
* xref:../../../support/remote_health_monitoring/about-remote-health-monitoring.adoc#about-remote-health-monitoring[About remote health monitoring]
```

File: installing/installing_azure_stack_hub/upi/installing-azure-stack-hub-user-infra.adoc [hunk 5]
```diff
@@ -46,7 +44,7 @@ include::modules/installation-azure-limits.adoc[leveloffset=+2]
 [role="_additional-resources"]
 .Additional resources
 
-* xref:../../scalability_and_performance/optimization/optimizing-storage.adoc#optimizing-storage[Optimizing storage].
+* xref:../../../scalability_and_performance/optimization/optimizing-storage.adoc#optimizing-storage[Optimizing storage].
```
Comment (adellape) [inline installing/installing_azure_stack_hub/upi/installing-azure-stack-hub-user-infra.adoc:47]:
```suggestion
* xref:../../../scalability_and_performance/optimization/optimizing-storage.adoc#optimizing-storage[Optimizing storage]
```

**Comment by brendan-daly-red-hat** [comment]
@adellape, Thanks for your review comments - I applied these suggestions. Could you take a look at these?

**Comment by brendan-daly-red-hat** [comment]
Suggested comments applied.

**Comment by jab-rh** [comment]
@brendan-daly-red-hat, looks like this doesn't merge cleanly into 4.17. That's kind of a problem as 4.17 and main ought to be mostly identical.

**Comment by bergerhoffer** [comment]
Looking!

**Comment by bergerhoffer** [comment]
@jab-rh @brendan-daly-red-hat I tracked it down to a revert of hiding WMCO docs that happened in 4.17 that didn't revert it exactly (it introduced some extra lines).

However, since this file is being deleted in this PR, you are fine to go ahead resolve this with a manual CP. After doing that, things should be in sync going forward.

@brendan-daly-red-hat please go ahead and create a manual cherry pick for this PR. 

As always, thanks for checking!

**Pull request merged**